app/controllers/api/v2/providers_controller.rb

```diff
@@ -107,6 +115,10 @@ def enrichment_params
             :region_code
           )
       end
+
+      def has_synced?
+        SearchAndCompareAPIService::Request.sync(@provider.courses)
```

Comment:
Don't think that will work...

One of those courses may not be in a ready state